### ISSUE TITLE
[BUG] Invalid DAG generated (EVGA 2080ti OC Black)

### ISSUE THREAD

**Issue by LeeDurant** [nicehash/NiceHashQuickMiner#352]
I have only been able to get quickminer to work a handful of times, and it seems the issue lies within the excavator miner itself or daggerhashimoto because the other miners in NiceHash Miner like TRex or NBminer seem to work just fine.

I have adjusted clocks higher and lower in both OCTuner and using Precision X1 to no avail. Only by chance am I able to get QuickMiner to work, where I then use the autotune setting and hope that it does not crash for a day or two. Since excavator and daggerhashimoto are the most efficient per kw/h on my system, I'd like to find what could make this work.

**Error**: `[0x000032c4][warning] cuda-daggerhashimoto Device #0 | Invalid DAG generated. Is your memory OC too high?`

**GPU**: EVGA 2080ti OC Black 12gb (11G-P4-2281) running driver version                                                               27.21.14.6589 (GeForce 465.89) non-DCH/UWD drivers.

**CPU**: AMD Ryzen 9 3900X

**Clocks and speeds**
Memory clock delta: 1100
Core clock max: 1080

**Version Affected:**
v.0.4.5.0, v.0.4.6.0 RC

**Current working solution**
Run NiceHash Miner without a miner active, start quickminer. Switch between the two until  share (green text) is accepted in quick miner, then close nicehash. Run efficiency or speed autotuner from there.

**Comment by nicehashdev** [comment]
Make sure there is no Afterburner or Precision X1 running in the background.

**Comment by LeeDurant** [comment]
Confirmed - uninstalled it, rebooted and confirmed nothing else was running like that in task manager.

**Comment by nicehashdev** [comment]
Enable logging and attach logs please.

**Comment by R77S77** [comment]
It's the same behaviour i reported last week with my RTX 2070 Super OC.

**Comment by nicehashdev** [comment]
Okay, I see that this bug is really there, but it affects small % of devices only. I believe it has something to do with memory allocations and drivers, but I need to find what exactly is going on and see what can be done to solve it.

**Comment by LeeDurant** [comment]
Logs attached from yesterday attached. It's been running overnight and most of today without issue, but I also haven't paused it / activated game mode since it started running properly. Avg 59 MH/s @ 410kH/J.

[exported_logs.zip](https://github.com/nicehash/NiceHashQuickMiner/files/6275088/exported_logs.zip)

**Comment by nicehashdev** [comment]
Is there any chance that you have your GPU PCIe set to Gen1 in BIOS?

**Comment by nicehashdev** [comment]
I was still not able to identify common source of the problem - it is just so sporadic. Can you give me some more information on what you did to get it fixed? Did you reinstall drivers? Used older drivers perhaps?

**Comment by LeeDurant** [comment]
I actually just rebuilt this machine a week ago, so the BIOS is untouched from stock - ASUS ROG Strix B550-I Gaming mITX.

As for getting things running again, I literally just flip between quick miner (QM) and nicehash miner (NHM) until I'm able to get a green share accepted in quick miner. From there, everything from the detailed stats on the web GUI to GUI-based optimization via dashboard or OCtune work like normal. It's only once I pause or enter game mode that things refuse to start again, prompting the fishing between QM and NHM.

**Comment by nicehashdev** [comment]
This sounds like a driver issue then. Could you try with older drivers?

**Comment by LeeDurant** [comment]
Any version in particular? I was having this issue, albeit less often, with my old build as well, so I'm not sure drivers are the whole story.

**Comment by nicehashdev** [comment]
Any after 460+ should be fine.

I am just releasing new version with updated CUDA; perhaps this will help.

**Comment by nicehashdev** [comment]
This is more and more turning into being a NVIDIA driver issue. People have reported that reinstalling QuickMiner (as changing Excavator - relocating it) can help. I believe that driver remembers some invalid settings - perhaps one that caused crash of the PC in one of previous sessions and these settings are being used when in fact should not have been. So there are two possible solutions:
1. Reinstall QuickMiner to another location so that NVIDIA driver tracks excavator.exe as a new application.
2. Reinstall NVIDIA driver (make sure to check "clean all").

I will leave this issue open for a while.

**Comment by LeeDurant** [comment]
Ok, sorry for delay - family stuff.

- Updated to 0.4.7.1 RC: no luck. 
- Uninstalled QM, reinstalled to new directory, restored files from windows protector and excluded folders: no luck for first several tries. After QM rebooted itself a good number of times, it restored itself to function with no outside intervention from myself. 

I ran the _efficiency_ OCtuner auto-tune and landed at the following clocks and values:
`[4/11/2021, 11:50:10 PM] Best OC for EFFICIENCY applied (memory: 6916MHz, core clock limit: 555MHz)`

I ran the _optimize-high_ auto-tune from the dashboard and landed the follow clocks and values:
`core clock max @ 1080,  memory clock @ 1100 for 143W @ 413 MH/J`
